src/GitHub.Exports/Models/RepositoryModel.cs

```diff
@@ -33,38 +31,22 @@ public class RepositoryModel : NotificationAwareObject, IRepositoryModel
             CloneUrl = cloneUrl;
         }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="RepositoryModel"/> class.
-        /// </summary>
-        /// <param name="path">
-        /// The path to the local repository from which repository name and clone URL will be
-        /// extracted.
-        /// </param>
-        /// <param name="gitService">The service used to find the repository's <see cref="Name"/> and <see cref="CloneUrl"/>.</param>
-        protected RepositoryModel(string path, IGitService gitService)
+        protected RepositoryModel()
         {
-            Guard.ArgumentNotNull(path, nameof(path));
-
-            var dir = new DirectoryInfo(path);
-            if (!dir.Exists)
-                throw new ArgumentException("Path does not exist", nameof(path));
-            var uri = gitService.GetUri(path);
-            Name = uri?.RepositoryName ?? dir.Name;
-            CloneUrl = gitService.GetUri(path);
         }
 
         /// <summary>
         /// Gets the name of the repository.
         /// </summary>
-        public string Name { get; }
+        public string Name { get; set; }
 
         /// <summary>
         /// Gets the repository clone URL.
         /// </summary>
         public UriString CloneUrl
```

Comment:
Future: should these properties be raising change notifications? If so, should other properties also be raising them (`CurrentBranch` for example).

Reply:
I think we should aim to get rid of the `Refresh` command so this model would be immutable after creation.

Reply:
I think we should aim to get rid of the `Refresh` command so this model would be immutable after creation.

Reply:
Sorry about the spam.  I'm having another go....

Reply:
Any another go.